tensorflow/core/platform/s3/s3_file_system.cc

```diff
@@ -252,19 +277,34 @@ class S3WritableFile : public WritableFile {
     if (!sync_needed_) {
       return Status::OK();
     }
-    Aws::S3::Model::PutObjectRequest putObjectRequest;
-    putObjectRequest.WithBucket(bucket_.c_str()).WithKey(object_.c_str());
     long offset = outfile_->tellp();
-    outfile_->seekg(0);
-    putObjectRequest.SetBody(outfile_);
-    putObjectRequest.SetContentLength(offset);
-    auto putObjectOutcome = this->s3_client_->PutObject(putObjectRequest);
+    std::shared_ptr<Aws::Transfer::TransferHandle> handle = 
+      transfer_manager_.get()->UploadFile(
+        outfile_, bucket_.c_str(), object_.c_str(),
+        "application/octet-stream", Aws::Map<Aws::String, Aws::String>());
+    handle->WaitUntilFinished();
+    int retries = 0;
+    while (handle->GetStatus() == Aws::Transfer::TransferStatus::FAILED &&
+           retries++ < kUploadRetries) {
+      // if multipart upload was used, only the failed parts will be re-sent
+      VLOG(1) << "Retrying Upload of s3://" << bucket_ << "/" << object_
```

Comment:
Let's use a constant instead of `1` in `VLOG`

Reply:
Just to clarify, do you want me to change 1 to a higher number? 

Or use a constant variable instead of the number 1? VLOG(1), VLOG(2), etc is actually consistent with usage in the current codebase. If we want to use a constant variable I don't see good options for constants which already exist (as the INFO, WARNING, ERROR constants would be unintuitive as they go in the opposite direction 0->1->2). I can create new constants specific to this file. 

Please let me know what do you want me to do.

Reply:
There should be some work to replace 0/1/2 with INFO/WARNING/ERROR to make these less unintuitive. I would prefer to use the names here but we can also merge this as it is and fix at a later time.